src/main/java/lavalink/client/io/LavalinkRestClient.java

```diff
@@ -0,0 +1,230 @@
+    }
+
+    private Consumer<JSONObject> consumeCallback(final AudioLoadResultHandler callback) {
+        return loadResult -> {
+            if (loadResult == null) {
+                callback.noMatches();
+                return;
+            }
+
+            try {
+                final String loadType = loadResult.getString("loadType");
+                final TrackLoadResultHandler trackLoadResultHandler = new TrackLoadResultHandler(loadResult);
+
+                switch (loadType) {
+                    case "TRACK_LOADED":
+                        callback.trackLoaded(trackLoadResultHandler.handleTrackLoaded());
+                        break;
+                    case "PLAYLIST_LOADED":
+                        callback.playlistLoaded(trackLoadResultHandler.handlePlaylistLoaded(false));
+                        break;
+                    case "NO_MATCHES":
+                        callback.noMatches();
+                        break;
+                    case "LOAD_FAILED":
+                        callback.loadFailed(trackLoadResultHandler.handleLoadFailed());
+                        break;
+                    case "SEARCH_RESULT":
+                        callback.playlistLoaded(trackLoadResultHandler.handlePlaylistLoaded(true));
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid loadType: " + loadType);
+                }
+            } catch (final Exception exception) {
+                callback.loadFailed(new FriendlyException(exception.getMessage(), FriendlyException.Severity.FAULT, exception));
+            }
+        };
+    }
+
+    private CompletableFuture<JSONObject> load(final String identifier) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                final String requestURL = buildBaseAddress(socket) + URLEncoder.encode(identifier, "UTF-8");
+                return apiGet(requestURL, socket.getPassword());
+            } catch (final Throwable exception) {
+                exception.printStackTrace();
+            }
+
+            return null;
+        });
+    }
+
+    private String buildBaseAddress(final LavalinkSocket socket) {
+        return socket.getRemoteUri().toString()
+                .replaceFirst("ws://", "http://")
+                .replaceFirst("wss://", "https://")
+                .concat("/loadtracks?identifier=");
+    }
+
+    private HttpClient buildClient() {
+        final HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        if (builderConsumer == null) return httpClientBuilder.build();
+
+        builderConsumer.accept(httpClientBuilder);
+        return httpClientBuilder.build();
+    }
+
+    private JSONObject apiGet(final String url, final String auth) throws IOException {
+        final HttpGet request = new HttpGet(url);
+        request.addHeader(HttpHeaders.AUTHORIZATION, auth);
+
+        final HttpClient httpClient = buildClient();
```

Comment:
I don't think we want to create a new client for each request

Reply:
Yeah, very true, sounds a bit redundant. I'll make it all use one client in the next update